test/JDBC/input/NUMERIC_TYPMOD.sql

```diff
@@ -0,0 +1,1231 @@
+        CASE 
+            WHEN 1=1
+            THEN CAST(1234.56 AS NUMERIC(10,2)) * cast(5.5 as money)
+            ELSE CAST(9876.54 AS DECIMAL(12,2)) / cast(6.43 as money)
+        END
+    ELSE CAST(0 AS DECIMAL(12,2))
+END AS result;
+GO
+
+SELECT CASE 
+    WHEN 1 = 1 THEN 
+        CASE 
+            WHEN 1=1
+            THEN CAST(1234.56 AS NUMERIC(10,2)) * cast(5.5 as smallmoney)
+            ELSE CAST(9876.54 AS DECIMAL(12,2)) / cast(6.43 as money)
+        END
+    ELSE CAST(0 AS DECIMAL(12,2))
+END AS result;
+GO
+
+
+-- TODO FIX: Output diff between TSQL and BBF.
+-- T_Param (Function parameter)
+DROP FUNCTION IF EXISTS dbo.TestParam
+GO
+
+CREATE FUNCTION dbo.TestParam (@param1 NUMERIC(10,4), @param2 DECIMAL(12,2))
+RETURNS TABLE
+AS
+RETURN
+(
+    SELECT CASE 
+        WHEN 1 = 1 THEN @param1 
+        ELSE @param2 
+    END AS result
+);
+GO
+
+SELECT * FROM dbo.TestParam(1234.5678, 9876.54);
+GO
+
+DROP FUNCTION dbo.TestParam
+GO
+
+-- T_Param (Function parameter)
+DROP FUNCTION IF EXISTS dbo.TestParam
+GO
+
+CREATE FUNCTION dbo.TestParam (@param1 NUMERIC(10,4), @param2 DECIMAL(12,2))
+RETURNS TABLE
+AS
+RETURN
+(
+    SELECT @param1 AS result
+    UNION ALL
+    SELECT @param2
+    ORDER BY result
+);
+GO
+
+SELECT * FROM dbo.TestParam(1234.5678, 9876.54);
+GO
+
+DROP FUNCTION dbo.TestParam
+GO
+
+-- T_NamedArgExpr (Named argument in a function call)
+DROP FUNCTION IF EXISTS dbo.TestNamedArg
+GO
+
+CREATE FUNCTION dbo.TestNamedArg (@param1 NUMERIC(10,4) = 0, @param2 DECIMAL(12,2) = 0)
+RETURNS TABLE
+AS
+RETURN
+(
+    SELECT @param1 AS result
+    UNION ALL
+    SELECT @param2
+    ORDER BY result
+);
+GO
+
+SELECT * FROM dbo.TestNamedArg(1234.56, 7890.1234);
+GO
+
+DROP FUNCTION dbo.TestNamedArg;
+GO
+
+-- Setup
+DROP FUNCTION IF EXISTS dbo.TestFunction;
+GO
+
+CREATE FUNCTION dbo.TestFunction (@num NUMERIC(10,2), @dec DECIMAL(12,2))
+RETURNS TABLE
+AS
+RETURN
+(
+    SELECT CASE 
+        WHEN @num > @dec THEN @num
+        ELSE @dec
+    END AS result
+);
+GO
+
+-- T_NamedArgExpr:
+-- Test Case 1
+SELECT * FROM dbo.TestFunction(123.45, 678.90);
+GO
+
+-- Test Case 2
+SELECT * FROM dbo.TestFunction(678.90, 123.45);
+GO
+
+-- Test Case 3
+DECLARE @n NUMERIC(10,2) = 123.45;
+DECLARE @d DECIMAL(12,2) = 678.90;
+SELECT * FROM dbo.TestFunction(@n, @d);
+GO
+
+-- Test Case 4
+SELECT * FROM dbo.TestFunction(678.90, CAST(123.45 AS NUMERIC(10,2)));
+GO
+
+-- Test Case 5
+SELECT * FROM dbo.TestFunction(CAST(123.45 AS NUMERIC(10,2)), CAST(678.90 AS DECIMAL(12,2)));
+GO
+
+DROP FUNCTION dbo.TestFunction;
+go
+
+
+
+-- TODO FIX: SQRT function returns float in TSQL. while
+-- selecting commontype between flaot ansd numeric,
+-- is selected. where as in BBF SQRT function returns numeric.
+-- Test Case: T_SubPlan with different numeric operations
+SELECT CASE 
+    WHEN (SELECT POWER(CAST(2 AS NUMERIC(5,2)), 3)) > 7
+    THEN (SELECT SQRT(CAST(100 AS NUMERIC(10,4))))
+    ELSE (SELECT CAST(PI() AS DECIMAL(10,8)))
+END AS result;
+GO
```

Comment:
these test cases are simple in terms of typmod. can you add cases where we can test boundaries of typmod? (38,0), (38,38), etc?